daemon/cancel_or_refund_test.go

```diff
@@ -56,10 +54,11 @@ func TestXMRTakerCancelOrRefundAfterKeyExchange(t *testing.T) {
 	aliceStatusCh, err := ac.TakeOfferAndSubscribe(makeResp.PeerID, makeResp.OfferID, providesAmt)
 	require.NoError(t, err)
 
-	var statusWG sync.WaitGroup
-	statusWG.Add(2)
+	var wg sync.WaitGroup
+	wg.Add(3)
 
 	go func() {
+		defer wg.Done()
 		for {
 			count, err := ec.PendingTransactionCount(ctx) //nolint:govet
 			require.NoError(t, err)
```

Comment:
We can't use `require` in a background go-process or the test won't fail for 30 minutes. Just converting to `assert` is also bad, because we don't want the code after it executed (we want the nested function to return).  There are multiple ways to fix it, but one way would be something like this:
```go
			if err != nil {
				t.Errorf("PendingTransactionCount failed: %s", err)
				return
			}
```

Reply:
updated!